src/test/kotlin/com/zepben/evolve/services/common/translator/TranslatorTestBase.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2021 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.services.common.translator
+
+import com.zepben.evolve.cim.iec61968.operations.OperationalRestriction
+import com.zepben.evolve.cim.iec61970.base.core.*
+import com.zepben.evolve.cim.iec61970.infiec61970.feeder.LvFeeder
+import com.zepben.evolve.database.sqlite.cim.CimDatabaseTables
+import com.zepben.evolve.database.sqlite.cim.tables.SqliteTable
+import com.zepben.evolve.database.sqlite.cim.tables.TableMetadataDataSources
+import com.zepben.evolve.database.sqlite.cim.tables.iec61970.base.core.TableNameTypes
+import com.zepben.evolve.database.sqlite.cim.tables.iec61970.base.core.TableNames
+import com.zepben.evolve.database.sqlite.common.TableVersion
+import com.zepben.evolve.services.common.BaseService
+import com.zepben.evolve.services.common.BaseServiceComparator
+import com.zepben.testutils.junit.SystemLogExtension
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import org.junit.jupiter.api.fail
+import kotlin.reflect.KClass
+import com.zepben.protobuf.cim.iec61970.base.core.NameType as PBNameType
+
+internal abstract class TranslatorTestBase<S : BaseService>(
```

Comment:
todo: Can you have a look at this and how I used it to make sure what I did was sane.

Reply:
Okay I think I understand what it's doing. I assume the converts all type will prevent the issue of forgetting to add to the validator since it'll actually error out now.